helpers/http_api.js

```diff
@@ -182,7 +182,7 @@ var middleware = {
 
 				if (
 					isNaN(value) ||
-					parseInt(value) != value ||
+					parseInt(value) !== Number(value) ||
```

Comment:
That's tricky, so we want to check here if the passed number is not a Number - then return just its value, but parse to Int if it's a number (mind that it's impossible now to pass decimal numbers through API - they get converted to Integers because of `parseInt`). 

I would rather use:
```
if (_.isNumber(value)) { 
   return +value; //Changes integer strings to integers, decimal numbers in strings to decimal numbers
}
return value;
```
@diego-G @nazarhussain please tell me what do you think.

Reply:
I got in trouble with this one also because the custom schema validator `parsedInt` is never used. Please @nazarhussain shed some light when you can.

Reply:
@yatki that statement doesn't work for every of these cases https://github.com/LiskHQ/lisk/blob/23fea4774d4baf53d4b904fbe77b49f2cc007680/test/unit/helpers/z_schema.js#L405

Reply:
So here we are checking if the value of some attributes are not (integers or integer formatted strings). Because by default with express every query param is treated as strings. 

So original values were with `radix=10`

```
isNaN(value) || parseInt(value) != value || isNaN(parseInt(value, radix))

isNaN(value) - Value of param is not a number
parseInt(value) != value - Parsed integer value not same as original 
isNaN(parseInt(value, radix)) - Parsing to integer with 10 base is not possible 
```

If any of above condition meet we want to preserve the value in the same format. e.g. If a `balance` query param came as string value of `123454` it will be converted to integer equivalent value. But if `balance` query param came as string value of `1234.54` it will left out as it is because `parseInt(value) != value` condition matches. 

And this was the exact reason we used `!=` instead of `!==`, so we can match strings directly with integers. So would be nice to actually do double conversion and match against same number. 

```
parseInt(value).toString() !== String(value)
```

Reply:
So, as query parameters, we only want to accept integers. No digits allowed as every value should come in Beddows. 

`parseInt(value).toString() !== String(value)` keeps the behaviour the same.